src/app/topic-overview/topic-overview/topic-overview.component.ts

```diff
@@ -0,0 +1,46 @@
+import {Component, OnDestroy, OnInit} from '@angular/core';
+import {TopicList} from '../../topic-widgets/topic-list/topic-list.component';
+import {TopicService} from '../topic-service/topic.service';
+import {ActivatedRoute} from '@angular/router';
+import 'rxjs/add/operator/filter';
+
+
+@Component({
+  selector: 'xgb-topic-overview',
+  templateUrl: './topic-overview.component.html'
+})
+export class TopicOverviewComponent implements OnInit, OnDestroy {
+
+  topicList: TopicList = new TopicList();
+
+  serverTitle: string;
+
+  constructor(private topicService: TopicService,
+              private route: ActivatedRoute) {
+  }
+
+  ngOnInit() {
+    this.topicService.getServerTitle().then((serverTitle) => {
+      this.serverTitle = serverTitle;
+    });
+    let subscription;
+
+    switch (this.route.snapshot.url[0].path) {
```

Comment:
I think it would be cleaner to use the [`data`-Property](https://angular.io/guide/router) or inheritance to solve this switching?  
I always get nervous when I see path components directly in code since the EPJ, we had many problems with that while refactoring & changing paths of components 🙈

Reply:
Agreed to use the data-property!